src/core/Events.js

```diff
@@ -183,16 +183,17 @@ L.Evented = L.Class.extend({
 			return this;
 		}
 
-		var handler = L.bind(function () {
-			this
-			    .off(types, fn, context)
-			    .off(types, handler, context);
-		}, this);
+		// remove the listener first to avoid potential recursion and then execute the function
+
+		var origFn = fn;
+		var _this = this;
+		fn = function () {

```

Comment:
ah no, OK, it comes from the arguments. But why override it instead of naming it `handler` like before?